discovery/main.go

```diff
@@ -0,0 +1,62 @@
+package discovery
+
+import (
+	"fmt"
+
+	"github.com/miekg/dns"
```

Comment:
is there a reason to not use the builtin https://golang.org/pkg/net/#LookupSRV? 

We aren't specifying a particular DNS server, just using the resolve.conf, meaning we gain nothing afaict

Reply:
TBH I googled around and found that library to be used in some places. I will take a look at your suggestion